smart-contracts/contracts/dex-router-osmosis/src/tests/initialize.rs

```diff
@@ -0,0 +1,483 @@
+                            Coin {
+                                denom: denom0.to_string(),
+                                amount: Uint128::from(1000000u128),
+                            }
+                            .into(),
+                        ),
+                    },
+                    PoolAsset {
+                        weight: "1".to_string(),
+                        token: Some(
+                            Coin {
+                                denom: denom1.to_string(),
+                                amount: Uint128::from(1000000u128),
+                            }
+                            .into(),
+                        ),
+                    },
+                ],
+                future_pool_governor: "overwritten".to_string(),
+            },
+            MsgCreateBalancerPool {
+                sender: "overwritten".to_string(),
+                pool_params: None,
+                pool_assets: vec![
+                    PoolAsset {
+                        weight: "1".to_string(),
+                        token: Some(
+                            Coin {
+                                denom: denom0.to_string(),
+                                amount: Uint128::from(1000000u128),
+                            }
+                            .into(),
+                        ),
+                    },
+                    PoolAsset {
+                        weight: "1".to_string(),
+                        token: Some(
+                            Coin {
+                                denom: denom2.to_string(),
+                                amount: Uint128::from(1000000u128),
+                            }
+                            .into(),
+                        ),
+                    },
+                ],
+                future_pool_governor: "overwritten".to_string(),
+            },
+            MsgCreateBalancerPool {
+                sender: "overwritten".to_string(),
+                pool_params: None,
+                pool_assets: vec![
+                    PoolAsset {
+                        weight: "1".to_string(),
+                        token: Some(
+                            Coin {
+                                denom: denom1.to_string(),
+                                amount: Uint128::from(1000000u128),
+                            }
+                            .into(),
+                        ),
+                    },
+                    PoolAsset {
+                        weight: "1".to_string(),
+                        token: Some(
+                            Coin {
+                                denom: denom2.to_string(),
+                                amount: Uint128::from(1000000u128),
+                            }
+                            .into(),
+                        ),
+                    },
+                ],
+                future_pool_governor: "overwritten".to_string(),
+            },
+        ],
+    )
+}
+
+pub fn init_test_contract(
```

Comment:
This fixture (as the others we have on cl-vault) looks very monolitic. Maybe we should start writing smaller helpers like create_cl_pool() as single one. create_gamm_pool() also as a single one. And another one to instantiate the contract which could probably be avoided as it would be a one-line invoke.

So in each test case we can create only the pool we need, and instantiate the contract accordingly to then do things such as SetPath, RemovePath and Swap. Swap will need liquidity so the admin should also create an initial position to facilitate the swaps in the happy cases we plan to write.